ddtrace/tracer/tracer_test.go

```diff
@@ -946,6 +946,28 @@ func TestTracerFlush(t *testing.T) {
 	})
 }
 
+func TestTracerReportsHostname(t *testing.T) {
```

Comment:
Would it make sense to have 2 sub-tests? One that makes sure it gets attached when the env. var. is set and one that makes sure it doesn't otherwise (or when it's `false`).

Reply:
Only marginally.
It might make sense to pick a single test that represents all the normal positive and negative assertions.
That way, each additional feature test can focus just on that feature instead of the normal case.

Reply:
As you wish, I still feel it would be good to have tests for both on and off. It can't hurt, and I don't think it's much effort either.